auth_token/models.py

```diff
@@ -101,11 +105,89 @@ class AnonymousToken:
     user_agent = None
     is_expired = True
     is_from_header = False
+    is_from_cookie = False
     active_takeover = None
     backend = None
+    allowed_cookie = False
+    allowed_header = False
 
     def save(self):
         raise NotImplementedError
 
     def delete(self):
         raise NotImplementedError
+
+
+class VerificationTokenManager(models.Manager):
+
+    def deactivate(self, obj, slug=None, key=None):
+        self.filter_active_tokens(obj, slug, key).update(is_active=False)
+
+    def deactivate_and_create(self, obj, slug=None, deactivate_old_tokens=True, expiration_in_minutes=24 * 60):
+        if deactivate_old_tokens:
+            self.deactivate(obj, slug)
+        return self.create(
+            content_type=ContentType.objects.get_for_model(obj.__class__),
+            object_id=obj.pk,
+            slug=slug,
+            expiration_in_minutes=expiration_in_minutes
+        )
+
+    def exists_valid(self, obj, verification_key, slug=None):
+        for token in self.filter_active_tokens(obj, slug):
+            if token.check_key(verification_key):
+                return True
+        return False
+
+    def filter_active_tokens(self, obj, slug=None, key=None):
+        qs = self.filter(
+            is_active=True,
+            slug=slug,
+            content_type=ContentType.objects.get_for_model(obj.__class__),
+            object_id=obj.pk
+        )
+        return qs.filter(key=key) if key else qs
+
+
+class VerificationToken(models.Model):
+    """
+    Specific verification tokens that can be send via e-mail to check user authorization (example password reset)
+    """
+
+    created_at = models.DateTimeField(auto_now_add=True, null=False, blank=False)
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.TextField()
+    content_object = GenericForeignKey('content_type', 'object_id')
+    key = models.CharField(null=False, blank=False, max_length=100)
```

Comment:
Tady jsme se bavili s Petrem že by to chtělo přidat constraint na unikátní key. A při generování kontrolovat zda už v DB není (stačí odchytit výjimku při uložení a přegenerovat klíč). Prakticky tady dovolouješ aby byly klíče duplikátní - což podle mě nemůže fungovat. Šance že se tak stane asi není velká, ale pokud jo, může to nadělat celkem neplechu.